src/main/java/core/basesyntax/service/impl/PurchaseTransactionHandler.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.FruitServiceImpl;
+
+public class PurchaseTransactionHandler implements TransactionHandler {
+    @Override
+    public void addTransaction(FruitTransaction fruitTransaction) {
+        FruitService fruitService = new FruitServiceImpl();
+        if (null != Storage.remnantsOfGoods.keySet().stream()
+                .filter(o -> o.equals(fruitTransaction.getFruit()))
+                .findFirst()
+                .orElse(null)) {
+            fruitService.updateFruit(Storage.remnantsOfGoods, fruitService
+                            .getFruit(fruitTransaction.getFruit()),
+                    fruitTransaction.getQuantity() * -1);

```

Comment:
it's very unclear logic. consider refactoring fruitService.updateFruit method to accept the exact quantity to set to the fruit.